src/utils.ts

```diff
@@ -173,6 +193,35 @@ export async function resolveLabel(label: string, config: Config): Promise<strin
   return config.provider.resolveName(`${label}.${config.registrar.domain}`);
 }
 
+export async function lookupAddress(address: string, did: string | null, config: Config): Promise<Lookup>  {
+  // First get eventual ENS Registration
+  let label: string | null = null;
+  try {
+    label = await config.provider.lookupAddress(address);
+  } catch (error) {
+    console.log("LookupAddress: Not able to get ENS Name, trying with IDX");
+  }
+  if (label && await config.provider.resolveName(label)) {
+    return {type: "ens", content: await getRegistration(label, config) as Registration };
+  }
+  // Else lookup IDX profile
+  if (did && isDid(did)) {
+    const content = await resolveIdxProfile(did, config);
+    if (content) return { type: "idx", content };
+  }
+  // If everything fails return address
+  return { type: "address", content: { name: address }};
+}
+
+// Resolves an existant idx profile
+export async function resolveIdxProfile(did: string, config: Config): Promise<radProfile | null> {
+  return config.idx.api.get<radProfile>("radProfile", did);
+}
+
+export function getCAIP10Address(address:string, protocol: string, impl: number): string {
+  return `${address}@${protocol}:${impl.toString()}`;
+}
```

Comment:
Functions like this are called `formatXXX` in the codebase, as it's clearer that no I/O is performed and no state is fetched.

Reply:
77762ea49c0d5ea4e8816b77921861ab52941991 solves this